### DISCUSSION TITLE
Can I use BufrDataProcess with IOSP2

### DISCUSSION THREAD

**Discussion by gtatarsky-IB** [Unidata/netcdf-java#1392] [Questions and Answers]
In the BufrDataProcess code, it appears this is hardcoded to attempt to open-in-memory with IOSP. However, what if I want to use IOSP2? Is this class even intended to be used? I'm looking to quickly and efficiently pull data from a BUFR IOSP2 file, and I'm getting a little confused, so any help would be appreciated.

**Comment by tdrwenski** [comment]
I am not sure if you should use that class-- it looks like it is nowhere used or tested in our code and indeed has the IOSP hardcoded. Unfortunately, not all of the BUFR code is well maintained.

If you want to read a BUFR file, I would suggest using our NetCDF API like:
```
NetcdfFile ncfile = NetcdfFiles.open(pathToYourFileAsStr)
```
which should work for all [CDM](https://docs.unidata.ucar.edu/netcdf-java/current/userguide/file_types.html) (common data model) file types, including BUFR. See also [our documentation here](https://docs.unidata.ucar.edu/netcdf-java/current/userguide/reading_cdm.html#opening-a-netcdffile).
